qemu/tests/timerdevice_boot.py

```diff
@@ -128,26 +140,42 @@ def verify_guest_clock_source(session, expected):
-    (host_time, guest_time) = utils_test.get_time(session, time_command,
-                                                  time_filter_re, time_format)
-    drift = abs(float(host_time) - float(guest_time))
-    if drift > timerdevice_drift_threshold:
-        raise error.TestFail("The guest's system time is different with"
-                             " host's. Host time: '%s', guest time:"
-                             " '%s'" % (host_time, guest_time))
-
-    get_hw_time_cmd = params.get("get_hw_time_cmd")
-    if get_hw_time_cmd:
-        error.context(
-            "Check the hardware time on guest and host", logging.info)
-        host_time = utils.system_output(get_hw_time_cmd)
-        guest_time = session.cmd(get_hw_time_cmd)
+    if params.get("rtc_base") == "2006-6-17":
+        guest_name = params["guest_name"]
+        if "rhel7" in guest_name:
+            error_context.context("Mask chronyd service")
+            cmd = "systemctl stop chronyd && systemctl mask chronyd"
+            session.cmd_output_safe(cmd)
+            error_context.context("Shoutdown the guest")
+            vm.destroy()
+            env.unregister_vm(vm.name)
+            error_context.context("Boot the guest", logging.info)
+            vm_name = params["main_vm"]
+            env_process.preprocess_vm(test, params, env, vm_name)
+            vm = env.get_vm(vm_name)
+            vm.verify_alive()
+            session = vm.wait_for_login(timeout=timeout)
+        verify_base_date_system_time(time_command, time_filter_re, time_format)
+    else:
+        error_context.context("Check the system time on guest and host", logging.info)
+        (host_time, guest_time) = utils_test.get_time(session, time_command,
+                                                      time_filter_re, time_format)
         drift = abs(float(host_time) - float(guest_time))
         if drift > timerdevice_drift_threshold:
-            raise error.TestFail("The guest's hardware time is different with"
-                                 " host's. Host time: '%s', guest time:"
-                                 " '%s'" % (host_time, guest_time))
+            test.fail("The guest's system time is different with"
+                      " host's. Host time: '%s', guest time:"
+                      " '%s'" % (host_time, guest_time))
+
+        get_hw_time_cmd = params.get("get_hw_time_cmd")
+        if get_hw_time_cmd:
+            error_context.context("Check the hardware time on guest and host", logging.info)
+            host_time = utils.system_output(get_hw_time_cmd)
+            guest_time = session.cmd(get_hw_time_cmd)
+            drift = abs(float(host_time) - float(guest_time))
+            if drift > timerdevice_drift_threshold:
+                test.fail("The guest's hardware time is different with"
+                          " host's. Host time: '%s', guest time:"
+                          " '%s'" % (host_time, guest_time))
```

Comment:
lines 169-178 are same as lines 206-216, a function is preferred. Thanks. :smile: